northd/northd.c

```diff
@@ -10772,13 +10772,25 @@ build_lrouter_nat_flows_for_lb(struct ovn_lb_vip *lb_vip,
             meter = copp_meter_get(COPP_REJECT, od->nbr->copp, meter_groups);
         }
 
+        int l3dgw_ports_index = 0;
         if (lb_vip->n_backends || !lb_vip->empty_backend_rej) {
+            for (size_t j = 0; j < od->n_l3dgw_ports; j++) {
+                if (find_lrp_member_ip(od->l3dgw_ports[j], lb_vip->vip_str)) {
+                    l3dgw_ports_index = j;
+                    break;
+                } else {
+                    static struct vlog_rate_limit rl = VLOG_RATE_LIMIT_INIT(5, 1);
+                    VLOG_WARN_RL(&rl,
+                                 "No matched LRP's subnet for VIP %s",
+                                 lb_vip->vip_str);
```

Comment:
Is your intention to log when no GW port has a subnet matching the VIP?  If so, this will log for every port that doesn't match the VIP and that seems wrong to me.